src/lib/services/RuleEngine.ts

```diff
@@ -50,3 +56,35 @@ export interface RuleEngine {
 	 */
 	isEngineRequested(filterValues: string[], engineOptions: Map<string, string>): boolean;
 }
+
+export abstract class AbstractRuleEngine implements RuleEngine {
+
+	abstract getName(): string;
+    abstract getTargetPatterns(): Promise<TargetPattern[]>;
+    abstract getCatalog(): Promise<Catalog>;
+    abstract shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): boolean;
+    abstract run(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): Promise<RuleResult[]>;
+    abstract init(): Promise<void>;
+	abstract matchPath(path: string): boolean;
+	abstract isEnabled(): Promise<boolean>;
+	abstract isEngineRequested(filterValues: string[], engineOptions: Map<string, string>): boolean;
+
+    async runHighLevel(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): Promise<RuleResult[]>{
+        const results = await this.run(ruleGroups, rules, target, engineOptions);
+        const normalizedResults = this.normalizeSeverity(results);
+        return normalizedResults;
+    }
+	
+	public async normalizeSeverity(results: RuleResult[]): Promise<RuleResult[]>{
+		for (let x=0; x<results.length; x++){
```

Comment:
Nitpick: Have a look at typescript's more modern For loop capabilities. Especially to loop through arrays: https://www.tutorialsteacher.com/typescript/for-loop